packages/ember-metal/lib/property_events.js

```diff
@@ -115,40 +115,36 @@ function notifyPropertyChange(obj, keyName, _meta) {
   }
 }
 
-let DID_SEEN;
+let DID_SEEN = new Map();
 
 // called whenever a property has just changed to update dependent keys
 function dependentKeysDidChange(obj, depKey, meta) {
   if (meta.isSourceDestroying() || !meta.hasDeps(depKey)) { return; }
   let seen = DID_SEEN;
-  let top = !seen;
+  let top = seen.length === 0;
 
   if (top) {
-    seen = DID_SEEN = {};
+    seen = DID_SEEN;
   }
 
   iterDeps(notifyPropertyChange, obj, depKey, seen, meta);
 
   if (top) {
-    DID_SEEN = null;
+    DID_SEEN.clear();
```

Comment:
I'm not totally sure if it is correct to share the instance because `dependentKeysDidChange` can be called recursively. If you're not totally sure yourself we should probably change the logic back to how it was.